### PR TITLE
Use the autoroute path instead of the json path when generating pager links for taxonomy terms.

### LINKED ISSUE #12870: Pager links for TaxonomyTerms uses the JsonPath instead of the AutoroutePath
Orchard Core version 1.5

### Describe the bug
This one's a little complicated but basically it has to do with displaying a taxonomy term. If there are more results than the page size then the pager is displayed as expected. However, the links in the pager use the JsonPath instead of the Autoroute Path.

### To Reproduce
This one requires a little bit of configuration but it's not too bad.

1. Create a new OC site using the blog recipe.
2. Define a new content type, name: GenreTerm, parts: TitlePart and AutoroutePart. On the AutoroutePart check "Manage contined item routes".
3. Create a new taxonomy, name: Genres, term content type: Genre Term, check "route contained items", add a term named "Fiction".
4. Define a new content type, name: Book, parts: TitlePart, fields: TaxonomyField using the Genre taxonomy.
5. Create two new content items of type Book. On both of them set the TaxonomyField to "Fiction".
6. Under Configuration > Settings > General set the page size to 1.

### Expected behavior
OK, so now we should have a row in the AutorouteIndex like this
image

If I manually enter genres/fiction in the address box I see this. So far so good. I see one content item, as well as a pager since there are two results but the page size is one. This is what I expect.

image

However, if I click the next button I see the correct content item results but I also see the less friendly JsonPath version of the link.
image

- What I expect the link to be: genres/fiction?after=638046645160426759
- What the link actually is: genres?jsonPath=TaxonomyPart.Terms%5B0%5D&after=638046645160426759

If I manually enter genres/fiction?after=638046645160426759 in the address bar I do end up on the second page, so I think the route itself works. But it would be ideal if the pager generated that friendly link instead of the JsonPath version of the link.
image

### PR DESCRIPTION
**Pull request by rjpowers10**
Fixes #12870 

Not sure this is the correct fix but it seems to get me what I want.

Before, the next button in the pager uses the jsonPath
image

After, the next button in the pager uses the path
image

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs [hunk 1]
```diff
@@ -41,11 +44,17 @@ public override Task<IDisplayResult> DisplayAsync(ContentItem model, BuildDispla
             {
                 return Task.FromResult<IDisplayResult>(Initialize<TermPartViewModel>("TermPart", async m =>
                 {
+                    // Provide the route data for the term to the pager so it generates the pager links
+                    // using the Autoroute path instead of the jsonPath.
+                    var termContentItemMetadata = await _contentManager.PopulateAspectAsync<ContentItemMetadata>(model);
```
Comment (rjpowers10) [inline src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs:49]: I probably shouldn't assume the term content item has the AutoroutePart. I thought about adding

if (model.Has<AutoroutePart>())
{
    // Do the route data stuff
}

but I couldn't access the AutoroutePart in this project.

**Comment by Skrypt** [comment]
Sorry need to find time these days. Made me a reviewer of that one.

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs (hunk 1 above)
Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs:49]: Please try https://github.com/OrchardCMS/OrchardCore/issues/12870#issuecomment-1343119061.

**Review by Piedone — CHANGES_REQUESTED** [review]
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

Is this something you'd like to revisit any time soon or should we close?

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs (hunk 1 above)
Comment (rjpowers10) [reply src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs:49]: It's been so long; I will try to refresh my memory and take another look.

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TermPartContentDriver.cs:49]: Thank you!

**Comment by Skrypt** [comment]
I'm questioning myself if it's not already fixed. Remember vaguely working on fixing these a while ago.

**Comment by Piedone** [comment]
It's not, still `jsonPath` is used.

**Pull request closed without merging**